tap_s3_csv/s3.py

```diff
@@ -134,9 +137,83 @@ def sample_file(config, table_spec, s3_path, sample_rate):
 
         current_row += 1
 
-    LOGGER.info("Sampled %s rows from %s",
-                sampled_row_count,
-                s3_path)
+    LOGGER.info("Sampled %s rows from %s", sampled_row_count, s3_path)
+
+
+def get_records_for_jsonl(s3_path, sample_rate, iterator):
+
+    current_row = 0
+    sampled_row_count = 0
+
+    for row in iterator:
+
+        if (current_row % sample_rate) == 0:
+            if (sampled_row_count % 200) == 0:
+                LOGGER.info("Sampled %s rows from %s",
+                            sampled_row_count, s3_path)
+            decoded_row = row.decode('utf-8')
+            if decoded_row.strip():
+                row = json.loads(decoded_row)
+            else:
+                current_row += 1
+                continue
+            sampled_row_count += 1
+            yield row
+
+        current_row += 1
+
+    LOGGER.info("Sampled %s rows from %s", sampled_row_count, s3_path)
+
+
+def check_key_properties_and_date_overrides_for_jsonl_file(table_spec, jsonl_sample_records, s3_path):
+
+    all_keys = set()
+    for record in jsonl_sample_records:
+        keys = record.keys()
+        all_keys.update(keys)
+
+    if table_spec.get('key_properties'):
+        key_properties = set(table_spec['key_properties'])
+        if not key_properties.issubset(all_keys):
+            raise Exception('JSONL file "{}" is missing required key_properties key: {}'
+                            .format(s3_path, key_properties - all_keys))
+
+    if table_spec.get('date_overrides'):
+        date_overrides = set(table_spec['date_overrides'])
+        if not date_overrides.issubset(all_keys):
+            raise Exception('JSONL file "{}" is missing date_overrides key: {}'
+                            .format(s3_path, date_overrides - all_keys))
+
+
+def sample_file(config, table_spec, s3_path, sample_rate):
+
+    file_handle = get_file_handle(config, s3_path)._raw_stream
+
+    extention = s3_path.split(".")[-1].lower()
+
+    records = []
+
+    if extention == "csv":
+        iterator = csv.get_row_iterator(
+            file_handle, table_spec)  # pylint:disable=protected-access
+        records = get_records_for_csv(s3_path, sample_rate, iterator)
+    elif extention == "jsonl":
+        iterator = file_handle
+        records = get_records_for_jsonl(
+            s3_path, sample_rate, iterator)
+        check_jsonl_sample_records, records = itertools.tee(
+            records)
+        jsonl_sample_records = list(check_jsonl_sample_records)
+        if len(jsonl_sample_records) == 0:
+            LOGGER.exception(
+                'No row sampled, Please check your JSONL file %s', s3_path)
+            raise Exception(
+                'No row sampled, Please check your JSONL file {}'.format(s3_path))
+        check_key_properties_and_date_overrides_for_jsonl_file(
+            table_spec, jsonl_sample_records, s3_path)
+    else:
+        pass
```

Comment:
@dbshah1212 Please add a logger message for unsupported extension

Reply:
Added log